jbmc/regression/generics/type_erasure/TestClass.java

```diff
@@ -0,0 +1,35 @@
+class Gen<T> {
+  public T t;
+}
+
+class IWrapper {
+  public int i;
+}
+
+class BWrapper {
+  public boolean b;
+}
+
+class TestClass {
+  static void testFunction() {
+    Gen<IWrapper> a = new Gen<>();
+    Gen b = a;
+    assert(b ==  a);
+    Gen<BWrapper> c;
+    c = b;
+    assert(c == b);
+    // below assertion is incorrectly labelled as failure
```

Comment:
Comment wasn't clear (will update) - but this actually fails both with and without the change.

Reply:
Well because the body of Object.equals for GOTO functions looks like

```
to_return = NONDET(_Bool);
return to_return
```

Will raise an issue as if two objects are actually the same it should work!

Reply:
Which is just a property of not using the any models for java.lang.Object - will see if I can find out how regression tests are meant to use the core models. Thanks for making me dig a bit!